src/mono/CMakeLists.txt

```diff
@@ -511,17 +508,9 @@ if ((HOST_MACCAT AND HOST_ARM64) OR (TARGET_MACCAT AND TARGET_ARM64))
   set(TARGET_APPLE_MOBILE 1)
 endif()
 
-# Decide if we need zlib, and if so whether we want the system zlib or the in-tree copy.
+# Decide if we need zlib-ng.
 if(NOT DISABLE_EMBEDDED_PDB OR NOT DISABLE_LOG_PROFILER_GZ)
-  if(INTERNAL_ZLIB)
-    # defines ZLIB_SOURCES
-    include(${CLR_SRC_NATIVE_DIR}/external/zlib.cmake)
-  else()
-    # if we're not on a platform where we use the in-tree zlib, require system zlib
-    include(${CLR_SRC_NATIVE_DIR}/libs/System.IO.Compression.Native/extra_libs.cmake)
-    set(Z_LIBS)
-    append_extra_compression_libs(Z_LIBS)
-  endif()
+    include(${CLR_SRC_NATIVE_DIR}/external/zlib-ng.cmake)
```

Comment:
@akoeplinger I will move this `include` to the very top, including it's `if` branch, of course. I think it might be confusing to have it all the way down here.

Reply:
Nevermind, this might not be a good idea.